cmd/alerts/list_alerts.go

```diff
@@ -0,0 +1,197 @@
+package alerts
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/backplane-cli/cmd/ocm-backplane/login"
+	"github.com/openshift/backplane-cli/pkg/cli/config"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	AccountNamespace string = "openshift-monitoring"
+	Alertprom string = "alertmanager-main"
+	ContainerName string = "alertmanager"
+	LocalHostUrl string = "http://localhost:9093"
+	PodName string = "alertmanager-main-0"
+)
+
+var levelCmd string
+
+type logCapture struct {
+	buffer bytes.Buffer
+}
+
+type alertCmd struct {
+	clusterID string
+	getLevel  string
+	//active    bool
+}
+
+func (capture *logCapture) GetStdOut() string {
+	return capture.buffer.String()
+}
+
+func (capture *logCapture) Write(p []byte) (n int, err error) {
+	a := string(p)
+	capture.buffer.WriteString(a)
+	return len(p), nil
+}
+
+//osdctl alerts list ${CLUSTERID} --level [warning, critical, firing, pending, all] --active bool 
+
+func NewCmdListAlerts() *cobra.Command {
+	alertCmd := &alertCmd{}
+	newCmd := &cobra.Command{
+		Use:               "list <cluster-id>",
+		Short:             "list alerts",
+		Long:              `Checks the alerts for the cluster`,
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			alertCmd.clusterID = args[0]
+			ListCheck(alertCmd)
+		},
+	}
+
+	newCmd.Flags().StringVarP(&alertCmd.getLevel, "level", "", "", "Alert level [warning, critical, firing, pending, all]")
+	//newCmd.Flags().BoolVar(&alertCmd.active, "active", false, "Show only active alerts")
+
+	return newCmd
+}
+
+func getLevel(level string) string{
+	switch level{
+	case "warning":
+		levelCmd = "warning"
+	case "critical":
+		levelCmd = "critical"
+	case "firing":
+		levelCmd = "firing"
+	case "pending":
+		levelCmd = "pending"
+	case "all":
+		levelCmd = "all" 
+	default:
+		log.Fatalf("Invalid alert level: %s\n", level)
+		return ""
+	} 
+	return levelCmd
+}
+
+func ListCheck(cmd *alertCmd) {
+	clusterID := cmd.clusterID
+	levelCmd := cmd.getLevel
+	//active := cmd.active
+
+	defer func() {
+		if err := recover(); err != nil {
+			log.Fatal("error : ", err)
+		}
+	}()
+
+	_, kubeconfig, clientset, err := GetKubeCli(clusterID)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output, err := getAlerts(kubeconfig, clientset, LocalHostUrl, levelCmd, PodName)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Println("Print information from all alert", output)
+
+}
+
+func GetKubeCli(clusterID string) (client.Client, *rest.Config, *kubernetes.Clientset, error) {
```

Comment:
I think this also exists in another piece of osdctl code - so might be another good candidate for a utility function that could be reused between multiple commands.

Reply:
utility code 
https://github.com/openshift/osdctl/pull/500/commits/2c49c094ad8077e3bf0bce597ba16f3dab524470#diff-75037fd13be730126f5b1e01769f6f07fb98f9e53bc4fbb2622b295d6ea7a933

@bergmannf